### ISSUE TITLE
CanonicalUPGD lifetime step/utility-age counters wrap around at INT32_MAX

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#334]
`CanonicalUPGD.update` increments the global optimizer step and safe-profile per-parameter utility ages with ordinary signed-int32 addition. At `2_147_483_647` (INT32_MAX), the next update wraps both counters to `-2_147_483_648`. The negative correction clock disables the EMA bias correction (`next_age > 0` becomes false for the affected code path), abruptly changes the protection gate, and can produce a NaN parameter under the `paper_global` equation.

confirmed directly:
```text
step_before: 2147483647
step_after: -2147483648
age_before: 2147483647
age_after: -2147483648
corrected_utility: 0.0
gate: 0.5
parameter_after: 2.0250000953674316
expected_saturated_gate: 0.7310585975646973
```

reachability: `CanonicalUPGD` is exported from both `alberta_framework` and `alberta_framework.core`. `alberta_framework/benchmarks/upgd_ipmnist.py` constructs it with `official_experiment_global`; callers can also instantiate the public class directly. The failure requires a very long-lived or restored-at-limit optimizer state, but is reachable with real finite training gradients — not a hardcoded-safe-only path.

fix incoming: a JAX-compatible saturating int32 increment (`value + jnp.minimum(increment, maximum - value)`) applied to both the global step clock and per-element utility-age clocks. Checked against the module's cited Elsayed & Mahmood ICLR 2024 equation and its pinned official-implementation profile comments — saturating preserves the EMA bias-correction limit exactly, since for any configured `utility_decay < 1` the correction is already numerically 1 at this lifetime, so an unboundedly-increasing integer clock can't change the intended equation past that point anyway.
